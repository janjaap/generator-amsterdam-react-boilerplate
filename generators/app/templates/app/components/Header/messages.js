import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
    defaultMessage: 'Logged in as',
  },
  log_in: {
    id: `${scope}.log_in`,
    defaultMessage: 'Log in',
  },
  log_out: {
    id: `${scope}.log_out`,
    defaultMessage: 'Log out',
  },
});
