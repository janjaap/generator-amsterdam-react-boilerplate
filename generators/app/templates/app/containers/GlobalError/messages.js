import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
  unknown_error: {
    id: `${scope}.unknown_error`,
  },
  server_error: {
    id: `${scope}.server_error`,
  },
  service_unavailable: {
    id: `${scope}.service_unavailable`,
  },
  resource_not_found: {
    id: `${scope}.resource_not_found`,
  },
  no_vbo_data_available: {
    id: `${scope}.no_vbo_data_available`,
  },
  login_failed: {
    id: `${scope}.login_failed`,
  },
  logout_failed: {
    id: `${scope}.logout_failed`,
  },
  authorize_failed: {
    id: `${scope}.authorize_failed`,
  },
  error_reporting_title: {
    id: `${scope}.error_reporting_title`,
  },
  error_reporting_subtitle: {
    id: `${scope}.error_reporting_subtitle`,
  },
  error_reporting_label_name: {
    id: `${scope}.error_reporting_label_name`,
  },
  error_reporting_label_email: {
    id: `${scope}.error_reporting_label_email`,
  },
  error_reporting_label_comments: {
    id: `${scope}.error_reporting_label_comments`,
  },
  error_reporting_label_close: {
    id: `${scope}.error_reporting_label_close`,
  },
  error_reporting_label_submit: {
    id: `${scope}.error_reporting_label_submit`,
  },
  error_reporting_success_message: {
    id: `${scope}.error_reporting_success_message`,
  },
  error_reporting_error_generic: {
    id: `${scope}.error_reporting_error_generic`,
  },
  error_reporting_error_form: {
    id: `${scope}.error_reporting_error_form`,
  },
});
