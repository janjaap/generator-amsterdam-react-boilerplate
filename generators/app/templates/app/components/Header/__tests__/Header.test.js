import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from '@datapunt/asc-ui';
import 'jest-styled-components';
import messages from '../../../translations/nl.json';

const intlProvider = new IntlProvider({ locale: 'nl', messages });
const { intl } = intlProvider.getChildContext();

describe('Header', () => {
  let Header;
  const OLD_ENV = process.env;

  beforeEach(() => {
    // eslint-disable-next-line
    Header = require('../index.js').default;
  });

  beforeAll(() => {
    process.env.HOST = 'localhost';
    process.env.PORT = 8080;
    process.env.HTTPS = true;
  });

  afterAll(() => {
    delete process.env.HTTPS;
    delete process.env.PORT;
    delete process.env.HOST;

    process.env = { ...OLD_ENV };
  });

  it('matches the snapshot', () => {
    const { container, rerender } = render(
      <ThemeProvider>
        <Header intl={intl} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();

    rerender(
      <ThemeProvider>
        <Header intl={intl} isAuthenticated />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render log in buttons', () => {
    const { getByText } = render(
      <ThemeProvider>
        <Header intl={intl} />
      </ThemeProvider>,
    );

    expect(getByText('Inloggen')).not.toBeUndefined();
  });

  it('should render a log out button', () => {
    const { getByText } = render(
      <ThemeProvider>
        <Header intl={intl} isAuthenticated />
      </ThemeProvider>,
    );

    expect(getByText('Uitloggen')).not.toBeUndefined();
  });

  it('should call the onLoginLogoutButtonClick handler', () => {
    const onLoginLogoutButtonClick = jest.fn();

    const { getByText } = render(
      <ThemeProvider>
        <Header intl={intl} onLoginLogoutButtonClick={onLoginLogoutButtonClick} />
      </ThemeProvider>,
    );

    fireEvent(
      getByText('Inloggen'),
      new MouseEvent('click', {
        bubbles: true,
      }),
    );

    expect(onLoginLogoutButtonClick).toHaveBeenCalledWith(expect.anything(), 'datapunt');
  });
});
