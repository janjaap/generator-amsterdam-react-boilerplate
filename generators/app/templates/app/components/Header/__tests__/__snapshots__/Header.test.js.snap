// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 15px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  max-width: none;
}

.c5:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c13 {
  color: #004699;
  display: inline-block;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  font-family: inherit;
  font-weight: 300;
  font-size: 1rem;
  -webkit-letter-spacing: -0.01562em;
  -moz-letter-spacing: -0.01562em;
  -ms-letter-spacing: -0.01562em;
  letter-spacing: -0.01562em;
  line-height: 1;
  font-style: normal;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c13:focus {
  outline: none;
  background-color: #FEC813;
}

.c13:hover {
  color: #ec0000;
}

.c13:hover {
  color: inherit;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c11 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("img");
}

.c9 {
  display: inline-block;
  height: 30px;
  width: 68px;
}

.c9:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 3px;
}

.c7 {
  height: 50px;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi',sans-serif;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 .c8 {
  margin-right: 10px;
}

.c1 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
}

.c1 .c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c3 {
  max-width: 1080px;
  margin: 0 auto;
}

.c3 a {
  line-height: normal;
}

.c3 h1 {
  padding: 0 !important;
}

.c3 h1 a {
  font-weight: normal !important;
}

.c0 {
  position: relative;
  z-index: 2;
}

.c16 {
  position: absolute;
  top: 5px;
  right: 20px;
  z-index: 101;
}

.c16 button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: 0;
  cursor: pointer;
}

.c16 button > * {
  vertical-align: middle;
  margin-right: 5px;
}

.c16 button:hover,
.c16 button:focus {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c16 button:hover svg,
.c16 button:focus svg {
  fill: #ec0000;
}

@media screen and (max-width:375px) {
  .c13 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c13 {
    font-size: 16px;
  }
}

@media screen and (min-width:1200.02px) {
  .c9 {
    height: 68px;
    width: 100px;
  }

  .c9 .c10 {
    background-image: url("img");
  }
}

@media screen and (min-width:1200.02px) {
  .c7 .c8 {
    margin-right: 40px;
  }
}

@media screen and (max-width:1200px) {
  .c1 {
    position: fixed;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }

  .c1 .c12 {
    margin-top: -2px;
  }
}

@media screen and (min-width:1200.02px) {
  .c1 {
    box-shadow: none;
    position: relative;
  }

  .c1:after {
    content: '';
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    height: 50px;
    margin-top: -50px;
    background-color: #F5F5F5;
  }

  .c1 .c4 {
    z-index: 100;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c1 .c2 {
    background-color: transparent;
  }

  .c1 .c12 {
    margin-top: 0px;
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    line-height: 38px;
    font-size: 30px;
  }

  .c1 .c6 {
    height: 78px;
    padding: 15px 0;
  }

  .c1 .c14 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-left: 25px;
    margin-right: 25px;
    height: 50px;
  }
}

@media (min-width:1200px) {
  .c0 {
    z-index: 0;
  }
}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
  .c16 button {
    height: 30px;
  }
}

@media (max-width:720px) {
  .c16 {
    right: 0;
  }

  .c16 .login-adw,
  .c16 span {
    display: none;
  }
}

<div>
  <div
    class="c0"
    data-testid="site-header"
  >
    <div
      class="c1"
    >
      <header
        class="c2 c3 c4 c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8 c9"
            href="https://localhost:8080/"
          >
            <span
              class="c10 c11"
            />
          </a>
          <a
            class="c12 c13"
            href="https://localhost:8080/"
          >
            Registraties
          </a>
        </h1>
        <div
          class="c14 c15"
        >
          <nav
            class="no-print c16"
          >
            <ul
              class="links horizontal"
            >
              <li>
                <button
                  class="login"
                  type="button"
                >
                  <img
                    focusable="false"
                    width="20"
                  />
                  <span>
                    Inloggen
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </header>
    </div>
  </div>
</div>
`;

exports[`Header matches the snapshot 2`] = `
<div>
  <div
    class="Header__HeaderWrapper-sc-15eqjjx-1 eSRqiv"
    data-testid="site-header"
  >
    <div
      class="sc-eNQAEJ gJJdWk"
    >
      <header
        class="sc-bxivhb Header__StyledHeader-sc-15eqjjx-0 egtpxo sc-chPdSV bHlKv"
      >
        <h1
          class="sc-jKJlTe eUAGBL"
        >
          <a
            class="sc-ckVGcZ dbIcbn"
            href="https://localhost:8080/"
          >
            <span
              class="sc-dxgOiQ kyqLhc"
            />
          </a>
          <a
            class="sc-kgoBCf sc-kGXeez ciQiUp"
            href="https://localhost:8080/"
          >
            Registraties
          </a>
        </h1>
        <div
          class="sc-kpOJdX eKiqBq"
        >
          <nav
            class="no-print Header__StyledNav-sc-15eqjjx-2 jTrQPp"
          >
            <ul
              class="links horizontal"
            >
              <li>
                <button
                  type="button"
                >
                  <img
                    focusable="false"
                    width="20"
                  />
                  <span>
                    Uitloggen
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </header>
    </div>
  </div>
</div>
`;
